### PR TITLE
KYLIN-4470 The user cannot log in kylin normally after being assigned…

### PR DESCRIPTION
**Pull request by zhangayqian**
… to a group

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java [hunk 1]
```diff
@@ -191,6 +191,8 @@ public ManagedUser save(@PathVariable("userName") String userName, @RequestBody
         }
         logger.info("Saving {}", user);
 
+        user.setPassword(pwdEncode(user.getPassword()));
```
Comment (shaofengshi) [inline server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java:194]: I'm not quite understand why it need reset the password. Could you add a description?

Comment (zhangayqian) [reply server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java:194]: Debugging found that the password of the ‘user’ passed in by this method is not encoded. If the password is directly updated to the metadata, the authentication of the user name and password will fail  for bad credential.

Comment (shaofengshi) [reply server-base/src/main/java/org/apache/kylin/rest/controller/UserController.java:194]: I see

**Review by shaofengshi — APPROVED** [review]
+1 LGTM

**Comment by zhangayqian** [comment]
### Check list
- [ ] Step 1. [**OPTIONAL**]Understand background/root cause/design basically(one hour EST) . Its issue type is 
    - [ ] bug fix
    - [x] new feature
    - [ ] enhancement
- [ ] Step 2. Test cases are designed and documented (30 minutes EST).
- [ ] Step 3. Prepare specific env, for example:
    - mock data (maybe some specific data type)
    - test env (maybe install a RDBMS instance).
- [ ] Step 4. Verify and make sure test cases passed.
- [ ] Step 5. Paste manual important steps and screenshots here (20 minutes EST). 
    - If you find difficulty in pick most important evidence, please attach diagnosis package. 
- [ ] Step 6. Do more check in user perspective (20 minutes EST)
    - [ ] Doc need be updated? And if it is updated? Ask help for release manager if so.
    - [ ] If it is a breaking change so we should notify Kylin community? Ask help for release manager if so.
- [ ] Step 7.  Summarize this test (20 minutes EST).

### Total estimate
- 4 hours for small issue from optimistic estimate. 

### Note
If you find some unexpected and **unrelated** error/mistake, please DO report it if it is truly a mistake, I think we may research and fix it in the future.

If you find background information/root cause analysis is not complete or ambiguous, please try to contact to author or do a quick research and record what you find. It is a good chance to learn something interesting.

If you find it is not easy to design testcase, please notify release manager.

**Comment by zhangayqian** [comment]
## Design test cases

### Prepare test env
- bin/kylin.sh start

### Test case
- Create user named `test_user` and assign it to `ROLE_ANALYST` group
- Sign out ADMIN and use `test_user` user login
### Expected result
- User `test_user` can login kylin normally

**Comment by zhangayqian** [comment]
### Prepare test env

- [x] Built on commit_SHA : 5d592b91769b5129c76df3c8b1deb3bd58ebcbb8
- [x] Hadoop env : CDH 5.7

**Comment by zhangayqian** [comment]
### Verify test case
- kylin 3.0.1
User `test_user` can't log in kylin for `Bad credentials` after being assign group.
image
- master
User `test_user` can log in kylin normally after being assign group.
image

**Comment by zhangayqian** [comment]
## Summary
- Bug fixed.

**Pull request merged**